lib/api/route/submessage.dart

```diff
@@ -0,0 +1,19 @@
+import 'dart:convert';
+
+import '../core.dart';
+import '../model/submessage.dart';
+
+/// https://zulip.readthedocs.io/en/latest/subsystems/widgets.html#polls-todo-lists-and-games
+Future<void> sendSubmessage(ApiConnection connection, {
+  required int messageId,
+  required SubmessageData content,
+}) {
+  return connection.post('sendSubmessage', (_) {}, 'submessage', {
+    'message_id': messageId,
+    // 'widget' applies to submessages for polls and todos.
+    // There is also 'zform' for trivia quiz bot, but we currently do not have
+    // plan to support that.
+    'msg_type': RawParameter('widget'),
+    'content': RawParameter(jsonEncode(content)),
```

Comment:
Would we get the same result from `'content': content` as we do from this, and if so, would it be preferable because it looks simpler?

Reply:
The case of `content` is a bit peculiar, because the API expects a plain string instead of an object, while the string is expected to be some JSON encoded value.

I think it would be good to mention this in comments.